### ISSUE TITLE
Notifications on comments

### ISSUE THREAD

**Issue by asimonw** [emfoundation/ce100-app#1380]
Currently, the poster of a challenge already gets a notification when someone posts a comment on their challenge. We'd like to extend this so that everyone who has posted a comment on a challenge gets "subscribed" to the thread for that specific challenge. 

This means that:
- If a user has previously posted on a challenge, they get a notification every time someone else comments on the same challenge.
- The user can opt out of this type of notification in their settings. **Question**: should this be a separate notification, or can it be grouped with "I'd like to be notified when someone comments on one of my challenges"?
- There will for the time being be no way to unsubscribe from individual challenge threads

To be added: fields to include in the notification, copy and mockup of email (including title)

**Comment by fannyBkane** [comment]
**Fields to include in the notification email:**

**Email:** info@cello.community
**Name of sender:** cello

**Subject line:** New comment on...(text TBC @Kimberley-Heath)

**Body of email:**
- name of person who receives the email
- name person who commented
- organisation the person who commented are from
- challenge title + link
- full comment text
- link to the corresponding challenge
**Email footer**
- text: why they’re receiving the email
- link to their personal settings

**Comment by fannyBkane** [comment]
> The user can opt out of this type of notification in their settings.
![screen shot 2018-06-12 at 11 04 21](https://user-images.githubusercontent.com/20577339/41284155-66424b6a-6e30-11e8-8b8c-386de01f2fdb.png)
[Link](https://invis.io/3DEU1V0A6#/299074282_My_personal_settings) to mobile prototype

**Comment by fannyBkane** [comment]
@sophielevens @Danwhy Please note that the form layout on the desktop prototype is different from the staging site. Don't update any styling to match the prototype, this will be addressed in issue #1125

**Comment by Kimberley-Heath** [comment]
@sophielevens @Danwhy Here is the email text; please note that I underlined words that are merge fields, although they do not need to be underlined in the final email. The pinks are links and if they could be shown in the cello pink #FA3265 (not underlined) in the final email that would be great! Although the last line "Too many emails? Change notification settings here" - please can this be the lavender grey and for the "here" link underlined #585667. Thanks!

Subject line: **Someone has also commented on the challenge "challenge name"**
Email body:
![screen shot 2018-06-13 at 10 36 18](https://user-images.githubusercontent.com/36198201/41343108-a48a8e3a-6ef5-11e8-838c-f745e3295de2.png)

Merge fields & links:
name of person who receives the email
name person who commented
organisation the person who commented are from
challenge title
full comment text
Challenge link
link to their personal settings

**Comment by Kimberley-Heath** [comment]
Hey, this works! However there is two small changes needed... 

- [ ] I posted a challenge, and commented on my own challenge. When someone else commented on my challenge I received two notifications - one was 'Someone has commented on **your** challenge...' which is right, the second was 'Someone has **also commented** on the challenge...' which should only send to those who have not posted the challenge so they only receive one notification.

- [ ] Sorry another style change as I have just got some advice off of a proofreader at work.. Whenever a challenge name is listed, please can it be italic, without speech marks/quotation marks in all cases? This means that the challenge name in the subject line (in the emails below) should be italic without speech marks as well. 

Thank you!

![screen shot 2018-06-25 at 11 55 02](https://user-images.githubusercontent.com/36198201/41846462-a0644f58-786e-11e8-9513-97644a8570ad.png)

**Comment by sophielevens** [comment]
Hi @Kimberley-Heath, 

Please see your requested changes in the screenshot below. 

![image](https://user-images.githubusercontent.com/25961258/41903975-2e86e820-792f-11e8-9024-f046ec2b32c2.png)

>  This means that the challenge name in the subject line (in the emails below) should be italic without speech marks for both emails.

Having done a bit of research I don't think it is possible for me to style the subject line of an email. I have, however, removed the quotation marks around the Challenge name as you asked. Let me know what you think 😄

**Comment by asimonw** [comment]
@sophielevens As I suggested in #1398 I think it's better to keep the quotation marks in the subject line, as otherwise it's not clear what part of the subject line is the title of the challenge.

@Kimberley-Heath do you agree?

**Comment by sophielevens** [comment]
Okay, 

I will wait for @Kimberley-Heath to get back to us on this 👍

**Comment by Kimberley-Heath** [comment]
Yes that is fine if it is the same for both, thanks!

**Comment by Kimberley-Heath** [comment]
Thanks @sophielevens! Both are consistent and I am now only receiving one notification. @asimonw would you mind also testing this?

**Comment by asimonw** [comment]
@sophielevens seems to work very well now, but still one tiny issue:

All the links in the notification emails are to `http` instead of `https`, causing the link-throughs to not work properly. Are you using the `ROOT_URL` var? In that case, I'd like to change that var to make sure that resolves it.

**Comment by sophielevens** [comment]
Yes, we're using the `ROOT_URL` var, @Danwhy has just changed it to https on Heroku

**Comment by asimonw** [comment]
@sophielevens Thanks, but unfortunately that didn't seem to solve the issue.

Every time I follow a link from a notification email, I'm asked to log in again (even when I'm already logged in in another tab as that same user). On top of that the redirect string on the login page seem to have no effect at all. This means that I always end up on the matches page, which makes the links in the notification email rather pointless..

So there seem to be two issues:
- I'm always required to log in again (strangely, this doesn't happen if I copy the link location and paste it into the browser url bar manually..)
- The redirect query on the login page has no effect anymore. This is a regression and would be great to fix in any case

**Comment by Kimberley-Heath** [comment]
@asimonw I always go straight to the page when I click the link without having to log in...
